test/framework/utilities.py

```diff
@@ -51,40 +51,43 @@
 from easybuild.tools.filetools import mkdir, read_file
 from easybuild.tools.module_naming_scheme import GENERAL_CLASS
 from easybuild.tools.modules import modules_tool
+from easybuild.tools.options import EasyBuildOptions
 
 
-class EnhancedTestCase(TestCase):
-    """Enhanced test case, provides extra functionality (e.g. an assertErrorRegex method)."""
+# make sure tests are robust against any non-default configuration settings;
+# involves ignoring any existing configuration files that are picked up, and cleaning the environment
+# this is tackled here rather than in suite.py, to make sure this is also done when test modules are ran separately
 
-    def assertErrorRegex(self, error, regex, call, *args, **kwargs):
-        """Convenience method to match regex with the expected error message"""
-        try:
-            call(*args, **kwargs)
-            str_kwargs = ', '.join(['='.join([k,str(v)]) for (k,v) in kwargs.items()])
-            str_args = ', '.join(map(str, args) + [str_kwargs])
-            self.assertTrue(False, "Expected errors with %s(%s) call should occur" % (call.__name__, str_args))
-        except error, err:
-            if hasattr(err, 'msg'):
-                msg = err.msg
-            elif hasattr(err, 'message'):
-                msg = err.message
-            elif hasattr(err, 'args'):  # KeyError in Python 2.4 only provides message via 'args' attribute
-                msg = err.args[0]
-            else:
-                msg = err
-            try:
-                msg = str(msg)
-            except UnicodeEncodeError:
-                msg = msg.encode('utf8', 'replace')
-            self.assertTrue(re.search(regex, msg), "Pattern '%s' is found in '%s'" % (regex, msg))
-            self.assertTrue(re.search(regex, msg), "Pattern '%s' is found in '%s'" % (regex, msg))
+# keep track of any $EASYBUILD_TEST_X environment variables
+test_env_var_prefix = 'EASYBUILD_TEST_'
+eb_test_env_vars = dict([(key, val) for (key, val) in os.environ.items() if key.startswith(test_env_var_prefix)])
+
+# clean up environment from unwanted $EASYBUILD_X env vars
+for key in os.environ.keys():
+    if key.startswith('EASYBUILD_'):
```

Comment:
on second thought, no, that won't work, since I'm redefining `$EASYBUILD_TEST_FOO` as `$EASYBUILD_FOO`, the `$EASYBUILD_TEST_FOO` shouldn't stay around, it should be undefined to make sure the tests are isolated from any haphazard config settings